lib/internal/fs.js

```diff
@@ -352,9 +352,12 @@ function validateLen(len) {
   if (!isInt32(len)) {
     if (typeof value !== 'number') {
       err = new ERR_INVALID_ARG_TYPE('len', 'number', len);
-    } else {
+    } else if (Number.isInteger(len)) {
       // TODO(BridgeAR): Improve this error message.
       err = new ERR_OUT_OF_RANGE('len', 'an integer', len);
+    } else {
+      // 2 ** 32 === 4294967296
+      err = new ERR_OUT_OF_RANGE('len', ' > -4294967296 && < 4294967296', len);
```

Comment:
`isInt32` is actually testing against a signed integer. So the range is not `2 ** 32 - 1` but `2 ** 31 - 1` instead.

Besides that the range should probably be >= 0, if I am not mistaken. The reason is that the length will be coerced to zero in case it is negative (in the call sites).

Reply:
I see `Marh.max(0,len)` always invoked after `validateLen`,  it seems that  this range should be >=0 but not in this function.

Besides, I think range error won't be throw after `isInt32` test ,is it right ?

Reply:
ok , I understand. thanks for your help , I will close it

Reply:
Oh, please feel free to keep this open. It was definitely the right direction and can probably land after addressing my comments and fixing the tests accordingly (I expect some tests might fail due to the changed error messages - and if not, we should add some).